spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetWriters.java

```diff
@@ -126,38 +126,39 @@ private SparkParquetWriters() {
     @Override
     public ParquetValueWriter<?> primitive(DataType sType, PrimitiveType primitive) {
       ColumnDescriptor desc = type.getColumnDescription(currentPath());
+      Type.ID id = primitive.getId();
```

Comment:
It is not clear the use of ID here.

Reply:
This is used to create `FieldMetrics` for individual writer so that we can later have a mapping between id and the new counter. Do you want me to add a comment in the code?

Reply:
I think it would be more clear with more documentation for the `FieldMetrics` class and the `Stream<FieldMetrics> metrics()` method.